mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -253,7 +257,14 @@ def SYCLGetRangeResult : AnyTypeOf<[I64, SYCL_RangeType]>;
 // CONSTRUCTOR OPERATION
 ////////////////////////////////////////////////////////////////////////////////
 
-def ConstructorArgs : AnyTypeOf<[SYCLMemref, IndexType, IntMemrefType, SYCL_IDType, SYCL_RangeType]>;
+def ConstructorArgs : AnyTypeOf<[SYCLMemref,
+                                 VectorEltTy,
+                                 IndexType,
+                                 VectorSplatArg,
```

Comment:
IntMemrefTy has been removed here and replaced by VectorSplatArg ... so we are more permissive now.  That is OK (assuming all constructors do really take those types) ... however I think it would be good to use more descriptive names for the few type aliases that this PR adds.

Reply:
I am fine with VectorElemTy. The VectorSplatArg is a memref of any vector element type. I guess there is some constructor in the SYCL runtime that takes that as an argument, correct?

Reply:
Yes, the vector "splat" constructor:

```cpp
explicit constexpr vec(const DataT& arg);
```

I don't like being so permissive either, but it's a compromise we have to take for now...

I'll resolve this conversation now.